src/lease/mod.rs

```
@@ -65,74 +60,99 @@ use std::sync::Arc;
+                        |e| panic!("Fail to send request, the error is {}", e)
+                    ),
+                    _ = shutdown_rx => return
+                };
             }
-        });
-
-        // monitor inbound watch response and transfer to the receiver
-        tokio::spawn(async move {
-            let mut shutdown_rx = shutdown_rx.fuse();
-            let mut inbound = futures::select! {
-                res = client.lease_keep_alive(request).fuse() => res.unwrap().into_inner(),
-                _ = shutdown_rx => { return; }
-            };
+        })
+        .detach();
 
+        smol::spawn(async move {
+            let mut shutdown_rx = shutdown_reponse.into_future().fuse();
             loop {
+                #[allow(clippy::mut_mut)]
                 let resp = futures::select! {
-                    resp = inbound.message().fuse() => resp,
-                    _ = shutdown_rx => { return; }
+                    resp_opt = client_resp_receiver.next().fuse() => resp_opt.unwrap_or_else(
+                        || panic!("Fail to receive client response")
+                    ),
+                    _ = shutdown_rx => return
                 };
                 match resp {
-                    Ok(Some(resp)) => {
-                        resp_sender.send(Ok(From::from(resp))).unwrap();
-                    }
-                    Ok(None) => {
-                        return;
+                    Ok(resp) => {
+                        resp_sender
+                            .send(Ok(From::from(resp)))
+                            .await
+                            .unwrap_or_else(|_| panic!());
                     }
                     Err(e) => {
-                        resp_sender.send(Err(From::from(e))).unwrap();
+                        resp_sender
+                            .send(Err(From::from(e)))
+                            .await
+                            .unwrap_or_else(|_| panic!());
```

no empty panic and don't ignore the error